vignettes/metadata.Rmd

```diff
@@ -0,0 +1,102 @@
+---
+title: "Adding metadata"
+author: "Thierry Onkelinx"
+output:
+  rmarkdown::html_vignette:
+        fig_caption: yes
+vignette: >
+  %\VignetteIndexEntry{Adding metadata}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`git2rdata` supports extra metadata since version 0.4.1.
+Metadata is stored in a separate file with the same name as the data file, but with the extension `.yml`.
+The metadata file is a YAML file with a specific structure.
+The metadata file contains a generic section and a section for each variable in the data file.
+The generic section contains information about the data file as a whole.
+The variable sections contain information about the variables in the data file.
+The metadata file is stored in the same directory as the data file.
+
+The generic section contains the following mandatory fields, automatically created by `git2rdata`:
+
+- `git2rdata`: the version of `git2rdata` used to create the metadata.
+- `datahash`: the hash of the data file.
+- `hash`: the hash of the metadata file.
+- `optimize`: a logical indicating whether the data file is optimized for `git2rdata`.
+- `sorting`: a character vector with the names of the variables in the data file.
+- `split_by`: a character vector with the names of the variables used to split the data file.
+- `NA string`: the string used to represent missing values in the data file.
+
+The generic section can contain the following optional fields:
+
+- `table name`: the name of the dataset.
+- `title`: the title of the dataset.
+- `description`: a description of the dataset.
+
+The variable sections contain the following mandatory fields, automatically created by `git2rdata`:
+
+- `type`: the type of the variable.
+- `class`: the class of the variable.
+- `levels`: the levels of the variable (for factors).
+- `index`: the index of the variable (for factors).
+- `NA string`: the string used to represent missing values in the variable.
+
+The variable sections can contain the following optional fields:
+
+- `description`: a description of the variable.
```

Comment:
`description` is called `field_descriptions` in frictionless

Reply:
The data package standard uses `description`. https://datapackage.org/standard/table-schema/

Reply:
Sorry, you are correct. I mistakenly took `field_descriptions` from a data package I had made, but that was not following the standard naming 🤔